docs/resource/README.md

```diff
@@ -102,7 +102,56 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
+**[1]:** MUST be unique for each instance of the same `service.namespace,service.name` pair
+(in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique).
+The ID helps to distinguish instances of the same service that exist at the same time
+(e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent
+and stay the same for the lifetime of the service instance, however it is acceptable that
+the ID is ephemeral and changes during important lifetime events for the service
+(e.g. service instance restarts).
+If the instance has no inherent unique ID that can be used as the value of this attribute
+it is recommended to generate a random Version 1 or Version 4
+[RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. Services aiming for reproducible UUIDs may also
+use Version 5. UUIDs are typically recommended, as we only need an opaque yet reproducible value for
+the purposes of identifying a service instance. Similar to what can be seen in the man page for the
+[`/etc/machine-id`](https://www.freedesktop.org/software/systemd/man/machine-id.html) file, the underlying data, such as pod name and namespace should be treated as
+confidential by this algorithm, being the user's choice to expose it or not via another resource attribute.
+
+When a UUID v5 is generated, the UUID's namespace MUST be set to:
+`${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`, so that the same service yields the
+same ID if the same value (`host.id`, `/etc/machine-id`, and so on) remains the same. It would still
+yield different results for different services on the same host.
+
+Users running their services on platforms such as Kubernetes are encouraged to explicitly set the
+`service.instance.id` using their existing automation, or by setting a value that can be used for a
+consistent value, such as `container.id`. Similarly, users of application servers such as `unicorn`
+are encouraged to set the `service.instance.id` on a per-worker basis.
+
+SDKs are required to use the following algorithm when generating `service.instance.id`:
+
+- If the user has provided a `service.instance.id`, via environment
+  variable, configuration or custom resource detection, this will
+  always be used and honored over generated IDs.
+- When any of the below combinations of resource attribute are provided, they should be used as the input
+  for generating a UUID v5. The values within each combination should be separated with dots:
+  * `container.id`
+  * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name`
+  * `host.id`
+- When the SDK is running in an environment where a `/etc/machine-id`
+  (see [MACHINE-ID(5)](https://www.freedesktop.org/software/systemd/man/machine-id.html))
+  is available, the machine-id should be used as the input for generating a UUID v5.
```

Comment:
So, ${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name} will be concatenated with one of these 4 and the result will be used as the "name" input to UUID v5 generation algo, right?

Reply:
Yes -- on my code example above, where you see "host.id", we'd have the machine ID instead.